operators/src/engine/result_descriptor.rs

```diff
@@ -21,6 +22,18 @@ use crate::util::Result;
 /// and spatial reference.
 pub trait ResultDescriptor: Clone + Serialize {
     type DataType;
+    type QueryRectangleSpatialBounds: AxisAlignedRectangle;
+    type QueryRectangleAttributeSelection: QueryAttributeSelection;
+
+    // Check the `query` against the `ResultDescriptor` and return `true` if the query is valid
+    // and `false` if, e.g., invalid attributes are specified
+    fn validate_query(
```

Comment:
I think if a query contains a band that doesn't exist, it should be invalid. Also it doesn't really make sense to ask the ResultDescriptor whether it can answer the query, because that's the queryprocessors job to do.